### ISSUE TITLE
InvalidArgumentError, Found Inf or NaN gradient(global norm).

### ISSUE THREAD

**Issue by CodeXiaoLingYun** [google/seq2seq#353]
NO,I am seeing the same error. I also used the same function(tf.clip_by_global_norm),but I found learning rate and function are not the key reasons. when i generate Vocab,i set the size is 4682,and the vocab_size is 4682 in train.py,too. as the same,i do not know whether decrease the batch size is useful.
I look at an answer that it might be related to the vanishing/exploding gradient? I do not have any methods.

In train.py
![image](https://user-images.githubusercontent.com/20829946/53682100-54f9f200-3d2d-11e9-9e44-bd04b858a102.png)

the error is
![image](https://user-images.githubusercontent.com/20829946/53682106-6642fe80-3d2d-11e9-8b4d-18fdd0ca6bc5.png)

and the error is found here
![image](https://user-images.githubusercontent.com/20829946/53682114-807cdc80-3d2d-11e9-9b2b-fde1c1816afd.png)

**Comment by CodeXiaoLingYun** [comment]
i found a questionable point is GPU:0. I think it may be related with my GPU,so i try to add the code like this
![image](https://user-images.githubusercontent.com/20829946/53682713-3c8cd600-3d33-11e9-8c55-1f6e9665d89f.png)
i do not know whether it is useful,i want to try.

**Comment by laosanli** [comment]
Have you solved the mistake
